File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(drops)
    solution = ""
    solution += "Pling" if drops % 3 == 0
    solution += "Plang" if drops % 5 == 0
    solution += "Plong" if drops % 7 == 0

    solution = "#{drops}" if solution == ""
    solution
  end
end

module BookKeeping
  VERSION = 3
end

File: javascript/raindrops/raindrops.js
var Raindrops = function() {};

Raindrops.prototype.convert = function(num) {
    solution_string = ""
    if (num % 3 === 0) {
        solution_string += "Pling"
    }
    if (num % 5 === 0) {
        solution_string += "Plang"
    }
    if (num % 7 === 0) {
        solution_string += "Plong"
    }
    if (num % 3 !== 0 && num % 5 !== 0 && num % 7 !== 0) {
        return String(num)
    }
    return solution_string
}

module.exports = Raindrops;